Fix part of #8668: Add jsdoc to story-editor-state.service.ts

pengcheng95 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #8668.
2. This PR does the following: Adds jsdoc to story-editor-state.service.ts

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

Commit:
Add jsdoc to story-editor-state.service.ts.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -193,6 +221,14 @@ angular.module('oppia').factory('StoryEditorStateService', [
+       * @param {function} successCallback - Function to be called if story
+       * publication status successfully changes.
+       * @return {bool} Whether or not the story publciation status
+       * succesfully changes.
```
typo: successfully

pengcheng95 replied:
Done.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -193,6 +221,14 @@ angular.module('oppia').factory('StoryEditorStateService', [
       },
 
+      /**
+       * @param {bool} newStoryStatusIsPublic - Whether or not the story is
+       * public or not.
```
"or not" in the end is redundant.

pengcheng95 replied:
Makes sense. Removed!

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -193,6 +221,14 @@ angular.module('oppia').factory('StoryEditorStateService', [
+       * public or not.
+       * @param {function} successCallback - Function to be called if story
+       * publication status successfully changes.
+       * @return {bool} Whether or not the story publciation status
```
typo: publication

pengcheng95 replied:
Done.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -94,29 +95,38 @@ angular.module('oppia').factory('StoryEditorStateService', [
       },
 
+      /**
+       * Set exploration Ids changed to false.
```
Maybe use the variable name like you did above.

pengcheng95 replied:
Updated!

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -94,29 +95,38 @@ angular.module('oppia').factory('StoryEditorStateService', [
       },
 
+      /**
+       * @return {boolean} Wether any of of the exporation ids have changed.
```
Typo 1: Whether
Typo 2: Redundant "of"
Typo 3: exploration

pengcheng95 replied:
Done.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -128,11 +138,15 @@ angular.module('oppia').factory('StoryEditorStateService', [
        * lifetime of the editor. This function never returns null, though it may
        * return an empty story object if the story has not yet been
        * loaded for this editor instance.
+       * @return {Story} Returns the current story to be shared.
```
Can we use the text above this instead?
```suggestion
       * @return {Story} The current story to be shared among the story
       * editor. Please note any changes to this story will be propogated
       * ..
       * ..
```

pengcheng95 replied:
Should I in the future just use more of the current description for the params and returns instead of having a description and than add a bit of context to params and returns?

kevintab95 replied:
Hm, that's a good question. I see both patterns in the codebase currently so I'm not quite sure what needs to be done in this case. Maybe @Showtim3 (the issue owner) can help.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -128,11 +138,15 @@ angular.module('oppia').factory('StoryEditorStateService', [
       },
 
+      /**
+       * @return {object} Returns an array of the list of skill summaries.
```
Hm, why do you think it is "an array of the list of" skill summaries and not a list of skill summaries? Also, is it possible to use the exact type instead of using "object" (we might have an interface defined for this type, PTAL)?

pengcheng95 replied:
What would you suggest is the best way to find an interface in this situation? Would it be best to run the application and search through there or do you have any tips? In this case, would the object be a SkillSummary? When you say to be more exact, instead of object should i say array of x?

kevintab95 replied:
Yep, in general, running the application and verifying the type (using console logs or debugger tool) would be a good way to do it. Also, you might find "Deducing variable type from the code" section in this [comment](https://github.com/oppia/oppia/issues/8668#issue-566518484) useful for your question. I think @nishantwrp would be able to give you more tips on this since he has been working on typing related stuff recently.

nishantwrp replied:
Hi @pengcheng95, yup you should try to deduce the type from code. The best way is to check what values are actually passed in this function. Looking at that you'll see that `EditableStoryBackendApiService` is used to fetch the story. And then looking at the backend controller you can easily find the correct type of skill summary. In this case which is `SkillSummaryBackendDict` in the `skill-summary-object.factory.ts`. (This file is not present in your branch, you'll need to merge from develop once.)

pengcheng95 replied:
Thanks for the tips. Updated!

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -218,6 +254,8 @@ angular.module('oppia').factory('StoryEditorStateService', [
       /**
        * Returns whether this service is currently attempting to save the
        * story maintained by this service.
```
Do we still need this part? Seems redundant.

pengcheng95 replied:
Removed.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -143,15 +157,24 @@ angular.module('oppia').factory('StoryEditorStateService', [
 
+      /**
+       * Returns the topic name stored within this service.
+       * @return {string} A string that represents the topic name.
```
```suggestion
       * @return {string} The topic name.
```

pengcheng95 replied:
Done.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -143,15 +157,24 @@ angular.module('oppia').factory('StoryEditorStateService', [
       },
 
+      /**
+       * Returns the topic name stored within this service.
```
Seems redundant. Can we remove it?

pengcheng95 replied:
Done.

kevintab95 commented on core/templates/pages/story-editor-page/services/story-editor-state.service.ts:
```diff
@@ -143,15 +157,24 @@ angular.module('oppia').factory('StoryEditorStateService', [
       },
 
+      /**
+       * Returns whether or not the story has been published.
```
Seems redundant, can we remove it?

pengcheng95 replied:
Done.

kevintab95 reviewed (changes requested):
Thanks @pengcheng95! I have left some comments. PTAL!

nishantwrp commented:
Hi @pengcheng95, please take a look at the review comments. Feel free to post any doubts or questions if any. Thanks

Commit:
Update jsdoc for story-editor-state.service.ts.

pengcheng95 commented:
Thanks for the comments Kevin! Updated the commit and added some questions.

iamprayush commented:
> Thanks for the comments Kevin! Updated the commit and added some questions.

Hey @pengcheng95, please make sure you assign the person you're expecting a response from. If you're requesting a review, make sure you "re-request review" from them in the reviewer's section, thanks!

kevintab95 commented:
Tagging @nishantwrp for [this](https://github.com/oppia/oppia/pull/9962#discussion_r461844212) comment.

Showtim3 commented:
It's fine to skip adding the overview of the function since we haven't standardized upon the convention and also it requires some context on why that function was created in the first place or what is its purpose.

Showtim3 commented:
Sorry about the last comment, lets add the function overview as well. PTAL [here](https://github.com/oppia/oppia/issues/8668#issue-566518484) for the updated descriptions.

pengcheng95 commented:
> > Thanks for the comments Kevin! Updated the commit and added some questions.
> 
> Hey @pengcheng95, please make sure you assign the person you're expecting a response from. If you're requesting a review, make sure you "re-request review" from them in the reviewer's section, thanks!

Ok will do!

pengcheng95 commented:
> Sorry about the last comment, lets add the function overview as well. PTAL [here](https://github.com/oppia/oppia/issues/8668#issue-566518484) for the updated descriptions.

When I look at one of the examples files that were linked, it looks like they don't have function overview. graph-input-rules.service.ts

kevintab95 commented:
Hi @pengcheng95 can you please resolve the merge conficts?
Also, @Showtim3 PTAL as codeowner!

ankita240796 commented:
Closing due to lack of response.
